### ISSUE TITLE
Exclude transitive dependencies?

### ISSUE THREAD

**Issue by y0ns0n** [johnrengelman/shadow#373]
### Shadow Version
2.0.2
### Gradle Version
4.5.1
### Expected Behavior
If I declare an artifact or configuration as non-transitive e.g. `compile ("com.zendesk:maxwell:1.8.0") { transitive = false }` I would expect the transitive dependencies to be excluded from the shadow jar.
### Actual Behavior
They are not excluded. It looks like the children of artifacts are always included?

**Comment by y0ns0n** [comment]
I was mistaken, `compile ("com.zendesk:maxwell:1.8.0") { transitive = false }` does work as expected, not including transitive dependencies.

However if I make the runtime configuration transitive, I must also specify the configuration in the shadowJar for the transitive property to be respected. Is that expected?

```
configurations {
    runtime {
        transitive = false
    }
}

shadowJar {
    configurations = [project.configurations.runtime]
}
```

**Comment by tylerbenson** [comment]
I also found the intricacies of this rather confusing.  Perhaps some better documentation explaining how this works would be useful.

**Comment by johnrengelman** [comment]
There are a lot of historical issue for this that have more information (like https://github.com/johnrengelman/shadow/issues/64)

tl;dr - you have to do the exclusions yourself, shadow provides various mechanisms for making this manageable (like configuring the “configurations” Property on the task, or using configurations.shadow). In the end, I’m not going to support it directly because there are decisions that shadow cant reason about correctly (for example of there is a common transitive dependency between 2 project dependencies where 1 is excluded)

**Comment by bucknerns** [comment]
This worked for me https://github.com/johnrengelman/shadow/issues/64#issuecomment-833180907
